Add explanation of IDE design process and include template (#61751)
dotnet/roslyn: docs/contributing/ide_design_process.md

# IDE Team Design Process

## Discussions / Issues

> The goal is that folks can create and discuss feature suggestions without needing to have verything fully figured out. Feel free to file a new feature suggestion and members of the oslyn-IDE team will add the "[Feature Request](https://github.com/dotnet/roslyn/issues?=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22Feature+Request%22)" label. However, hese feature suggestions won't be discussed until we have a proposal written for them.

Reviewer: oslyn-IDE :)

Reviewer 2: ```suggestion
The goal is that folks can create and discuss feature suggestions without needing to have everything fully figured out. Feel free to file a new feature suggestion and members of the Roslyn-IDE team will add the "[Feature Request](https://github.com/dotnet/roslyn/issues?=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22Feature+Request%22)" label. However, these feature suggestions won't be discussed until we have a proposal written for them.
```

> The goal is that folks can create and discuss feature suggestions without needing to have everything fully figured out. Feel free to file a new feature suggestion and members of the Roslyn-IDE team will add the "[Feature Request](https://github.com/dotnet/roslyn/issues?=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22Feature+Request%22)" label. However, these feature suggestions won't be discussed until we have a proposal written for them.

Reviewer 3: `folks` feels a bit like slang here, maybe contributors would be a better term?

> The goal is that folks can create and discuss feature suggestions without needing to have everything fully figured out. Feel free to file a new feature suggestion and members of the Roslyn-IDE team will add the "[Feature Request](https://github.com/dotnet/roslyn/issues?=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22Feature+Request%22)" label. However, these feature suggestions won't be discussed until we have a proposal written for them.

Reviewer 3: You say `a proposal written` here, but then refer to multiple different types of proposals below (both feature and design). Which did you mean here? Consider clarifying the difference between these types of proposals.

## Proposals

### Adding a proposal

> For a feature proposal to be reviewed a design proposal needs be appended to the github ssue. It should have the following information:

Reviewer 2: ```suggestion
For a feature proposal to be reviewed a design proposal needs be appended to the github issue. It should have the following information:
```

> For a feature proposal to be reviewed a design proposal needs be appended to the github issue. It should have the following information:

Reviewer 4: ```suggestion
For a feature proposal to be reviewed, a design proposal needs be appended to the GitHub issue. It should have the following information:
```

> - **Summary**: Brief summary of what this proposal is about

Reviewer 2: Other lines had ending punctuation.

```suggestion
- **Summary**: Brief summary of what this proposal is about.
```

> - **Background and Motivation**: What the problem is we are solving and in what context did ou encounter it?

Reviewer 2: ```suggestion
- **Background and Motivation**: What the problem is we are solving and in what context did you encounter it?
```

- **Proposed Feature**: How the proposed solution solves the problem

> - **Alternative Designs**: Were there other options you considered? How does this compare to eatures in other editors?

Reviewer 2: ```suggestion
- **Alternative Designs**: Were there other options you considered? How does this compare to features in other editors?
```

### Getting the issue championed by the IDE team

> A member of [Roslyn-IDE](https://github.com/orgs/dotnet/teams/roslyn-ide) will need to decide to "champion" that feature proposal and add it to the "On Deck" tab in the [IDE Design Review](ttps://github.com/dotnet/roslyn/projects/40) project. Behind the scenes the IDE Team tries to meet once a week to discuss these item.

Reviewer 2: ```suggestion
A member of [Roslyn-IDE](https://github.com/orgs/dotnet/teams/roslyn-ide) will need to decide to "champion" that feature proposal and add it to the "On Deck" tab in the [IDE Design Review](ttps://github.com/dotnet/roslyn/projects/40) project. Behind the scenes the IDE Team tries to meet once a week to discuss these issues.
```

> A member of [Roslyn-IDE](https://github.com/orgs/dotnet/teams/roslyn-ide) will need to decide to "champion" that feature proposal and add it to the "On Deck" tab in the [IDE Design Review](ttps://github.com/dotnet/roslyn/projects/40) project. Behind the scenes the IDE Team tries to meet once a week to discuss these issues.

Reviewer 4: ```suggestion
A member of [Roslyn-IDE](https://github.com/orgs/dotnet/teams/roslyn-ide) will need to decide to "champion" that feature proposal and add it to the "On Deck" tab in the [IDE Design Review](https://github.com/dotnet/roslyn/projects/40) project. Behind the scenes the IDE Team tries to meet once a week to discuss these issues.
```

> A member of [Roslyn-IDE](https://github.com/orgs/dotnet/teams/roslyn-ide) will need to decide to "champion" that feature proposal and add it to the "On Deck" tab in the [IDE Design Review](https://github.com/dotnet/roslyn/projects/40) project. Behind the scenes the IDE Team tries to meet once a week to discuss these issues.

Reviewer 3: I wouldn't use `tries` here. I would just state that we meet once a week, and use `generally` as a qualifier if you think one is necessary. There are exceptions, of course, but that is our general rule.
